app/views/tag/show/_nav_tabs.html.erb

```diff
@@ -1,44 +1,46 @@
-<ul class="nav nav-tabs" style="margin-top:10px;">
+<div class="dropdown" style="margin-top:10px; float:right;">
+  <a class="btn btn-outline-secondary dropdown-toggle" id="dropdownMenuButton" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
+  By Type
+  </a>
   <!-- when will it not be show...? -->
-  <% if params[:action] == "show" %>
-      <% unless params[:id].match("question:") %><li class="nav-item"><a class="nav-link <% if @node_type == "note" %> active<% end %>" href="/tag/<%= params[:id] %>"><i class="fa fa-file"></i> <span class="d-none d-lg-inline"><%= raw t('tag.show.research_notes') %></span></a></li><% end %>
-    <!-- can we create a generic node type hash, and give it href and class and content classes? -->
-    <!-- this should be an if/else, because @node_type should only be one value -->
-    <li class="nav-item"><a class="nav-link <% if @node_type == "questions" %> active<% end %>" href="/questions/tag/<%= params[:id] %>"><i class="fa fa-question-circle"></i> <span class="d-none d-lg-inline"><%= t('tag.show.questions') %></span></a></li>
-    <li class="nav-item"><a class="nav-link <% if @node_type=="wiki" %> active <% end %>" href="/wiki/tag/<%= params[:id] %>"><i class="fa fa-book"></i> <span class="d-none d-lg-inline"><%= raw t('tag.show.wiki_pages') %></span></a></li>
-    <li class="nav-item"><a class="nav-link <% if @node_type == "maps" %> active <% end %>" href="/maps/tag/<%= params[:id] %>"><i class="fa fa-map-marker"></i> <span class="d-none d-lg-inline"><%= t('tag.show.maps') %></span></a></li>
-
-    <!-- show if we can see the list of contributors
-    again, can we have a generic list of contributors that is
-    calculated in the controller? -->
-    <% if !@wildcard %>
-        <li class="nav-item"><a class="nav-link <% if @node_type == "contributors" %> active <% end %>" href="/contributors/<%= params[:id] %>"><i class="fa fa-user"></i> <span class="d-none d-lg-inline"><%= raw t('tag.show.contributors') %></span><span class="badge"><%=@length %></span></a></li>
+  <div class="dropdown-menu dropdown-menu-right mt-1" aria-labelledby="dropdownMenuButton">
+    <% if params[:action] == "show" %>
+        <% unless params[:id].match("question:") %><a class="dropdown-item <% if @node_type == "note" %> active<% end %>" href="/tag/<%= params[:id] %>"><i class="fa fa-file"></i> <span class="d-none d-lg-inline"><%= raw t('tag.show.research_notes') %></span></a><% end %>
```

Comment:
Can u use the dev console elements inspector to see what the code looks like? I mean, here, it seems like we should be able to test against it with `css_select`. Maybe `css_select` docs say that the element has to be shown?

If you can't get it to work, you could write a system test for it, where you literally send a click event to open the dropdown. I've done that for some system tests, so it's a backup plan!

Reply:
But Doesn't `css_select` work only when the element is being shown not when it's hidden like dropdown menu here ? 
I did system test though, to have a screenshot when menu is open and closed.